doc/grpc_release_schedule.md

```diff
@@ -0,0 +1,22 @@
+# gRPC Release Schedule
+
+Below is the release schedule for gRPC [Java](https://github.com/grpc/grpc-java/releases), [Go](https://github.com/grpc/grpc-go/releases) and [Core](https://github.com/grpc/grpc/releases) and its  dependent languages C++, C#, Objective-C, PHP, Python and Ruby.
+
+Releases are scheduled every six weeks on Tuesdays on a best effort basis. In some unavoidable situations a release may be delayed or a language may skip a release altogether and do the next release to catch up with other languages. See the past releases in the links above.
```

Comment:
nit: it may be good to interesting to mention the motivation behind six-weekly release schedule (e.g. good balance of delivering new features and fixes quickly to the users and not having too much release overhead, and explaing why we chose fixed release schedule over variable one)

Reply:
Done. Don't want to make it too verbose so skipped the rationale for fixed vs. variable release.